coding/coding_tests/mem_file_reader_test.cpp

```diff
@@ -24,3 +24,25 @@ UNIT_TEST(MemReaderSimple)
       TEST_EQUAL(c[j], i + j < n - 2? data[i + j + 1] : uint8_t(0xff), (i, j, n));
   }
 }
+
+UNIT_TEST(MemReaderStringView)
+{
+  std::string_view constexpr data = "123";
+  MemReader const memReader(data);
+  size_t const n = data.size();
+  for (size_t i = 0; i < n; ++i)
+  {
+    uint8_t c[n] = {0};
+    ReadFromPos(memReader, i, c, n - i);
+    for (size_t j = 0; j < n; ++j)
+      TEST_EQUAL(c[j], i + j < n ? data[i + j] : uint8_t{0}, (i, j, n));
+  }
+  MemReader const subReader = memReader.SubReader(1, 2);
+  for (size_t i = 1; i < n - 1; ++i)
+  {
+    uint8_t c[n] = {0};
+    ReadFromPos(subReader, i, c, n - i - 2);
```

Comment:
Strange test, honestly saying ..
i == 1 only
size == 3-1-2 == 0
subReader from i == 1 - is its end

Reply:
Fixed.